test/disjointFixedMallocPool.cpp

```diff
@@ -0,0 +1,521 @@
+        p1 = p2;
+        p2 = t;
+    }
+
+    // free + alloc first block
+    // the block should be reused
+    // currently there is no purging, so the alloc size shouldn't change
+    // there should be no block merging between used and not-used blocks
+    umf_result_t res = umfPoolFree(pool, p1);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).used_size, 2 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).alloc_size, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 3);
+
+    p1 = umfPoolMalloc(pool, 2 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).used_size, 4 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 3);
+
+    // free all allocs
+    // overall alloc size shouldn't change
+    // block p2 should merge with the prev free block p1
+    // and the remaining init block
+    res = umfPoolFree(pool, p1);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 3);
+    res = umfPoolFree(pool, p2);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).used_size, 0 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).alloc_size, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 1);
+
+    // alloc whole buffer
+    // after this, there should be one single block
+    p1 = umfPoolMalloc(pool, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).used_size, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).alloc_size, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 1);
+
+    // free all memory
+    // alloc 2 MB block - the init block should be splitted
+    res = umfPoolFree(pool, p1);
+    p1 = umfPoolMalloc(pool, 2 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).used_size, 2 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).alloc_size, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 2);
+
+    // alloc additional 2 MB
+    // the non-used block should be used
+    p2 = umfPoolMalloc(pool, 2 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).used_size, 4 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).alloc_size, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 3);
+    ASSERT_NE(p1, p2);
+
+    // make sure that p1 < p2
+    if (p1 > p2) {
+        void *t = p1;
+        p1 = p2;
+        p2 = t;
+    }
+
+    // free blocks in order: p2, p1
+    // block p1 should merge with the next block p2
+    // swap pointers to get p1 < p2
+    umfPoolFree(pool, p2);
+    umfPoolFree(pool, p1);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).used_size, 0 * MB);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).alloc_size, init_buffer_size);
+    ASSERT_EQ(umfFixedMemoryProviderGetStats(pp).blocks_num, 1);
+
+    // alloc 10x 2 MB
+    void *t[10] = {0};
```

Comment:
Make a constexpr int from the magic value of `10`.

Reply:
done